### ISSUE TITLE
Manual partitioning is really unfriendly

### ISSUE THREAD

**Issue by lnicola** [archlinux/archinstall#2444]
1. every partition starts the the beginning of the drive, instead of right after the previous partition

![image](https://github.com/archlinux/archinstall/assets/308347/d72f5450-083d-4fa3-bd3e-141f164a733b)

![image](https://github.com/archlinux/archinstall/assets/308347/cf4f0a67-1df8-44ec-8006-a0c85c3635e9)

![image](https://github.com/archlinux/archinstall/assets/308347/1a6c95da-26d3-4505-96ff-3b99d61570b2)

Notice how the two partitions overlap. You can spot this if you're paying attention, but you have to go back to figure out where the previous partition ends, then write down the sector number and try again.

2. you can't specify a partition size (relative end). So if you make a:
 - 0 start / 1 GB end
 - 1 GB start / 10 GB end
 - 10 GB start / 50 GB end

You get a 1 GB, a 9 GB and a 40 GB partition. `gdisk` supports both `10G` (ends at 10 GB) and `+10G` (has size of 10 GB).

https://github.com/archlinux/archinstall/issues/1213 suggests this is actually the size, but it doesn't match my experience.

3. the "format" option disappears after toggling it and it's not clear when it's going to format the partition or not
4. this might be intentional, but if you delete an existing partition, it gets marked as deleted; then if you delete it again, it disappears from the list. Now if you try to create a new partition, you can't use the whole disk. I don't have a screenshot unfortumately.

**Comment by Jazztache** [comment]
I second this. I am attempting to install Arch with manual partitioning right now, and it fails even with the best-fit partition scheme. It's consistently frustrating to deal with this and it would be a huge help if what is described above was implemented.

**Comment by lnicola** [comment]
@Jazztache I suggest using `gdisk` instead, and formatting the partitions. I did manage to name the installer work (with existing partitions), but it took me about four tries to get right.

**Comment by Jazztache** [comment]
> @Jazztache I suggest using `gdisk` instead, and formatting the partitions. I did manage to name the installer work (with existing partitions), but it took me about four tries to get right.

Oh yeah it always takes me multiple tries to get Arch running by messing with the partitions.
Gonna read the gdisk documentation to see if you can specify mountpoints.

**Comment by lnicola** [comment]
You can create the partitions using `gdisk` (or `parted`), format them, then specify the mount points in `archinstall`. That part seems to work fine.

**Comment by Jazztache** [comment]
Just tested this out. It thinks my third partition is encrypted somehow. I cannot get into the partitions screen of my disk encryption...

**Comment by arlsdk** [comment]
For manual partitioning, you could use **cfdisk**. More user-friendly and easier.

**Comment by HappyGoFishing** [comment]
Yeah cfdisk is the easiest command line partitioning tool.

**Comment by lnicola** [comment]
I mean, we're not discussing about the best non-GUI partitioning tool (`gdisk` is good enough for me), we're discussing what I believe to be a couple of bugs in `arch-install`.

**Comment by alexng353** [comment]
I agree. The Arch install partition manager is very difficult to use, and there aren't any good online tools to help you configure it properly. I was considering making an online partition helper, but I lack a lot of experience in partition management. That lack of experience also makes installing Arch a lot more difficult.

When I created my current partitions, which are weird sizes, I just used some rough math to make it happen.

Does anyone know any guides for making partitions with gdisk and then mounting them with archinstall?

**Comment by tony-sol** [comment]
Also, “Suggest partition layout” suggests overlapped partitions by default e.g.

40-**409640** /boot
**409640**-**12345678** /
**12345678**-123456789

And there’s no way to change them except delete and create manually

**Comment by codefiles** [comment]
@tony-sol
archinstall uses pyparted for partitioning, that is the standard behavior of parted. Here is a deeper dive into it.

The steps to create a `1GiB` ESP and root partition of remaining space with parted following the [GPT examples](https://wiki.archlinux.org/title/Parted#UEFI/GPT_examples) on the ArchWiki would be:

```
(parted) mkpart "EFI system partition" fat32 1MiB 1025MiB
(parted) set 1 esp on
(parted) mkpart "root partition" ext4 1025MiB 100%
(parted) type 2 4F68BCE3-E8CD-4DB1-96E7-FBCAF984B709
```

Notice that the ESP ends on `1025MiB` and the root partition starts on `1025MiB`. This is how all of the example partitions are shown on the ArchWiki, starting at the end of the previous.

Inspecting a `20GiB` disk that has been partitioned with these steps by using the command <code>parted -s <i>disk</i> unit s print free</code> will show the following table.

```
Number  Start      End        Size       File system  Name                  Flags
        34s        2047s      2014s      Free Space
 1      2048s      2099199s   2097152s                EFI system partition  boot, esp
 2      2099200s   41940991s  39841792s               root partition
        41940992s  41943006s  2015s      Free Space
```

There is no overlap, and the free space at the beginning (an [alignment buffer](https://wiki.archlinux.org/title/Advanced_Format#Partition_alignment)) and end are where the GPT header and backup header will reside, respectively.

Using _Suggest partition layout_ with the same disk to again create an ESP and root partition will yield:

```
Number  Start      End        Size       File system  Name  Flags
        34s        2047s      2014s      Free Space
 1      2048s      2099199s   2097152s   fat32              boot, esp
 2      2099200s   41940991s  39841792s  ext4
        41940992s  41943006s  2015s      Free Space
```

This is the same but with the partitions formatted. No overlap, working as intended.
